--- nx-X11/programs/Xserver/hw/nxagent/Keystroke.c
@@ -121,16 +121,6 @@
 
         break;
       }
-      case XK_m:
-      case XK_M:
-      {
-        if (nxagentOption(Rootless) == False)
-        {
-          *result = doMinimize;
-        }
-
-        break;
-      }
       case XK_Left:
       case XK_KP_Left:
       {
@@ -203,77 +193,6 @@
 
         break;
       }
-
-      case XK_J:
-      case XK_j:
-      {
-        nxagentForceSynchronization = 1;
-
-        return 1;
-      }
-
-      #ifdef DUMP
-
-      case XK_A:
-      case XK_a:
-      {
-        /*
-         * Used to test the lazy encoding.
-         */
-
-        nxagentRegionsOnScreen();
-
-        return 1;
-      }
-
-      #endif
-
-      #ifdef NX_DEBUG_INPUT
-
-      case XK_X:
-      case XK_x:
-      {
-        /*
-         * Used to test the input devices state.
-         */
-
-        if (X -> type == KeyPress)
-        {
-          if (nxagentDebugInputDevices == 0)
-          {
-            fprintf(stderr, "Info: Turning input devices debug ON.\n");
-    
-            nxagentDebugInputDevices = 1;
-          }
-          else
-          {
-            fprintf(stderr, "Info: Turning input devices debug OFF.\n");
-    
-            nxagentDebugInputDevices = 0;
-    
-            nxagentLastInputDevicesDumpTime = 0;
-          }
-        }
-
-        return 1;
-      }
-
-      case XK_Y:
-      case XK_y:
-      {
-        /*
-         * Used to deactivate input devices grab.
-         */
-
-        if (X -> type == KeyPress)
-        {
-          nxagentDeactivateInputDevicesGrabs();
-        }
-
-        return 1;
-      }
-
-      #endif
     }
   }
   else if ((X -> state & nxagentAltMetaMask) &&
